ingestion-sink/src/main/java/com/mozilla/telemetry/ingestion/sink/config/SinkConfig.java

```diff
@@ -116,23 +121,28 @@
   // BigQuery Load API limits maximum load requests per table per day to 1,000 but mixed mode
   // expects fewer than 1,000 messages per day to need file loads, so use streaming max delay.
   private static final String DEFAULT_STREAMING_LOAD_MAX_DELAY = DEFAULT_STREAMING_BATCH_MAX_DELAY;
+  // BatchExceptions don't count toward this limit. Only used when error output is configured,
+  // otherwise messages will be immediately nacked on failure.
+  private static final int DEFAULT_OUTPUT_MAX_ATTEMPTS = 3;
 
   @VisibleForTesting
   protected static class Output implements Function<PubsubMessage, CompletableFuture<Void>> {
 
-    private final Env env;
     private final OutputType type;
     private final Function<PubsubMessage, CompletableFuture<Void>> write;
 
-    private Output(Env env, OutputType outputType,
-        Function<PubsubMessage, CompletableFuture<Void>> write) {
-      this.env = env;
+    private Output(OutputType outputType, Function<PubsubMessage, CompletableFuture<Void>> write) {
```

Comment:
`env` has been split up into input/output/error, and a side-effect is that the `env` used in `getOutput` no longer needs to be kept around for use by `getInput`